packages/deployments/contracts/contracts/lib/Connext/ConnextLogic.sol

```diff
@@ -875,6 +898,33 @@ library ConnextLogic {
     bytes32 _transferId,
     bool _reconciled
   ) private {
+
+    // Is the domain sponsor
+    if (address(_args.sponsorVault) != address(0)) {
+      // fast liquidity path
+      if (!_reconciled) {
+        // Vault will return the amount of the fee they sponsored in the native fee
+        // NOTE: some considerations here around fee on transfer tokens and ensuring
+        // there are no malicious `Vaults` that do not transfer the correct amount. Should likely do a
+        // balance read about it
+
+        uint256 starting = IERC20Upgradeable(_adopted).balanceOf(address(this));
+        uint256 sponsored = _args.sponsorVault.reimburseLiquidityFees(_adopted, _args.executeArgs.amount);
```

Comment:
Have we addressed/discussed what happens if devs do *not* want to sponsor liq fees but *do* want to sponsor relayer fees?

Reply:
Obviously, this would return 0 right?

But that would be really gas inefficient, what with doing this contract call to `reimburseLiquidityFees` and then a bunch of useless addition.

We could have a mapping, however, that the sponsor vault owner could update in `setSponsorVault` that indicates whether to reimburse liquidity fees or not. Would technically add the cost of accessing storage in the use case where we *are* reimbursing both fees, but access lists in the contract call could alleviate that....